support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -230,4 +231,20 @@ object CreateZuoraSubscription {
     createdRequestId__c = state.requestId.toString,
     autoPay = state.paymentMethod.isLeft
   )
+
+  def updateRedemptionCode(
+    paymentMethod: Either[PaymentMethod, RedemptionData],
+    subscriptionData: SubscriptionData,
+    setCodeStatus: SetCodeStatus
+  ): Future[Unit] =
+    paymentMethod match {
+      case Right(rd: RedemptionData) =>
+        setCodeStatus(
+          RedemptionCode(rd.redemptionCode).right.get,
+          RedemptionTable.AvailableField.CodeIsUsed
+        )
+      case Left(_: PaymentMethod) if (subscriptionData.subscription.redemptionCode.isEmpty) => Future.successful(())
```

Comment:
Explain this if statement to me 😄

Reply:
For a paid sub, we should not be planning to put anything in the redemption code field of zuora.   That logic is hidden away in the builders.
If that all matches, then we can continue without updating the DB.
If there's a mismatch, then we throw an error (like a runtime assertion)

However if it needs explaining then it's not clear enough.  Having looked again, I wonder if it makes sense to call the "updateRedemptionCode" at all, if there is no redemption code.  Assuming we think the check adds value, we should have it as a separate step.

Having said that, I raised the separate refactor PR so that I could add the test for this (which is at the bottom of this PR) so maybe the value of the check is low now compared with the cost.